xacro/__init__.py

```diff
@@ -1026,14 +1047,16 @@ def main():
             sys.exit(2)  # gracefully exit with error condition
 
     # special output mode
-    if opts.just_deps:
-        out.write(" ".join(set(all_includes)))
+    #if opts.just_deps:
+    if opts['just_deps']:
+        out.write(' '.join(set(all_includes)))
         print()
-        return
+        return ' '.join(set(all_includes))
```

Comment:
Why do you want to create the output twice (here as a return value) and in line 1052 as output to `out`.
Same applies for the way more costly `doc.toprettyxml` below.
I think, as a user you should decide whether to use the python function interface and use the return value or call the xacro script and use the output generated.

Reply:
I could think of a use case to use the output from within a python launch script. But that no longer makes sense. I think that the best thing is to force the user to write output to a file when called from a script instead of returning the output.

Reply:
If you are forcing the user to write an output file, he can use xacro the old way, i.e. launch a sub process, as well. I thought, the main goal of this PR was to enable direct calling of `xacro.process()` and to utilize its _return value_ in the launch script. So, I expected that you would have removed the file output, but not the return value. 
I have implemented this idea in the last amended commit.
Please clarify your intent.

Reply:
" direct calling of xacro.process() and to utilize its return value in the launch script" sounds like the way to go! Thanks